### PR TITLE
[irods/irods#6286] Updated REPF documentation (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**
Cherry-picked from #200.

Updated REP links to point to 4-2-stable.

I think everything else is identical to main, but I'll do one last pass to make sure everything still applies.

### CODE REVIEW

**Comment by korydraughn** [comment]
Everything is correct.

However, the REPF docs do not explain what happens when a call fails before reaching the RULE_ENGINE_CONTINUE code.

I'm going to add a statement to the code example that makes it clear. I'll backport it to main once merged.

**Comment by trel** [comment]
Sounds good.  Good catch.

File: docs/plugins/pluggable_rule_engine.md [hunk 1]
```diff
@@ -92,48 +90,55 @@ Consider the following `server_config.json` with the accompanying `example.re`,
...
 def pep_resource_close_pre(rule_args, callback, rei):
-    callback.writeLine("serverLog","XXXX - core.py")
+    callback.writeLine("serverLog", "XXXX - core.py")
+
+    # Instruct the REPF to let the next rule engine plugin handle this PEP.
+    # If we remove the line below, the PEP in core.re will not execute. If an
+    # error occurs before reaching the line below, processing stops and the
+    # error is returned to the client. No other rule engines will be allowed
+    # to process the PEP.
+    return RULE_ENGINE_CONTINUE
```
Comment (korydraughn) [inline docs/plugins/pluggable_rule_engine.md:105]: I've added more words to the code comment to explain what happens if an error occurs before reaching the continuation code.

Thoughts?

File: docs/plugins/pluggable_rule_engine.md [hunk 2]
```diff
@@ -98,7 +98,10 @@ def pep_resource_close_pre(rule_args, callback, rei):
     callback.writeLine("serverLog", "XXXX - core.py")
 
     # Instruct the REPF to let the next rule engine plugin handle this PEP.
-    # If we remove the line below, the PEP in core.re will not execute.
+    # If we remove the line below, the PEP in core.re will not execute. If an
```
Comment (trel) [inline docs/plugins/pluggable_rule_engine.md:101]: "the PEP" -> "neither matching PEP"

File: docs/plugins/pluggable_rule_engine.md [hunk 3]
```diff
@@ -98,7 +98,10 @@ def pep_resource_close_pre(rule_args, callback, rei):
...
+    # error occurs before reaching the line below, processing stops and the
+    # error is returned to the client. No other rule engines will be allowed
+    # to process the PEP.
```
Comment (trel) [inline docs/plugins/pluggable_rule_engine.md:104]: "the PEP" -> "this PEP"

File: docs/plugins/pluggable_rule_engine.md (hunk 2 above)
Comment (korydraughn) [reply docs/plugins/pluggable_rule_engine.md:101]: "neither matching PEP in core.re will not execute." reads very weird to me.

Should "not" be removed?

Comment (trel) [reply docs/plugins/pluggable_rule_engine.md:101]: correct - take out the "not"

File: docs/plugins/pluggable_rule_engine.md (hunk 3 above)
Comment (korydraughn) [reply docs/plugins/pluggable_rule_engine.md]: done

File: docs/plugins/pluggable_rule_engine.md (hunk 2 above)
Comment (korydraughn) [reply docs/plugins/pluggable_rule_engine.md]: done

**Comment by trel** [comment]
looks good, squash

**Comment by korydraughn** [comment]
Squashed.

I think it's ready for pounds. Opening PR for main with latest tweak to code example.

**Comment by trel** [comment]
confirmed - please # all

**Comment by korydraughn** [comment]
Added #'s

**Pull request merged**